src/Donation/DonationRequestUtils.php

```diff
@@ -52,9 +53,15 @@ public function __construct(ServiceLocator $locator)
     /**
      * @throws InvalidDonationPayloadException
      */
-    public function createFromRequest(Request $request, float $amount, int $duration, ?Adherent $currentUser): DonationRequest
+    public function createFromRequest(Request $request, ?Adherent $currentUser): DonationRequest
     {
         $clientIp = $request->getClientIp();
+        $amount = (float) $request->query->get('montant');
+        $duration = $request->query->getInt('abonnement', PayboxPaymentSubscription::NONE);
+
+        if (!PayboxPaymentSubscription::isValid($duration)) {
```

Comment:
this is form validation and it already works with `PayboxSubscriptionValidator` and `@Assert` in `DonationRequest`
```
/**
     * @AssertPayboxSubscription
     */
    private $duration;
```

Reply:
I just move code from controller to this function.

And I think the goal was to throw an exception if the duration is not valid and not render the form.